packages/cloud_firestore/lib/src/firestore.dart

```diff
@@ -32,6 +35,11 @@ class Firestore {
           this,
         );
         _documentObservers[call.arguments['handle']].add(snapshot);
+      } else if (call.method == 'DoTransaction') {
+        final int transactionId = call.arguments['transactionId'];
+        return _transactionHandlers[transactionId](new Transaction(transactionId)).then<Map<String, dynamic>>((Map<String, dynamic> result) {
```

Comment:
This is a very long line, I would split it by assigning the new transaction and the handler to a variable.

Can you help me understand what `.then((result) => result)` does? Would it work if you left that part off, since the handlers already return a `Future`?

If you want to get fancy here, before returning the result you could flag the `Transaction` object as dead by calling some sort of private `_dispose()` method on the Dart `Transaction` object and then `assert(!disposed);` in all the methods of the `Transaction` are called. This would ensure that clients of the API don't store and try to reuse old `Transaction` objects outside of the handler. This is totally optional though.

Reply:
Indeed this line is unnecessarily long. I'll just return the result of then TransactionHandler.